integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/clients/jmsclient/JMSQueueMessageProducer.java

```
@@ -0,0 +1,209 @@
+    /**
+     * Method to send a BytesMessage.
+     *
+     * @param payload content of the BytesMessage to be sent
+     * @throws JMSException if an error occurs sending the BytesMessage
+     */
+    public void sendBytesMessage(byte[] payload) throws JMSException {
+        checkIfConnected();
+        BytesMessage bytesMessage = session.createBytesMessage();
+        bytesMessage.writeBytes(payload);
+        producer.send(bytesMessage);
+    }
+
+    /**
+     * Method to send a StreamMessage.
+     *
+     * @param payload content of the StreamMessage to be sent
+     * @throws JMSException if an error occurs sending the BytesMessage
+     */
+    public void sendStreamMessage(byte[] payload) throws JMSException {
+        checkIfConnected();
+        StreamMessage streamMessage = session.createStreamMessage();
+        streamMessage.writeBytes(payload);
+        producer.send(streamMessage);
+    }
+
+    /**
+     * Method to send a MapMessage.
+     *
+     * @throws JMSException if an error occurs sending the MapMessage
+     */
+    public void sendMapMessage() throws JMSException {
+        checkIfConnected();
+        MapMessage mapMessage = session.createMapMessage();
+        producer.send(mapMessage);
+    }
+
+    /**
+     * Method to send an ObjectMessage.
+     *
+     * @param payload content of the ObjectMessage to be sent
+     * @throws JMSException if an error occurs sending the ObjectMessage
+     */
+    public void sendObjectMessage(Serializable payload) throws JMSException {
+        checkIfConnected();
+        ObjectMessage objectMessage = session.createObjectMessage(payload);
+        producer.send(objectMessage);
+    }
+
+    /**
+     * Private method to check if session or producer are null, implying a connection was not established.
+     *
+     * @throws JMSException if a connection has not been established
+     */
+    private void checkIfConnected() throws JMSException {
+        if (session == null || producer == null) {
```

Do we really need this check? Ideally, the underline client implementation should throw an exception is the connection is not there.

---

better to avoid from RuntimeException by checking them before executing. So I think this would be better to have a null check

---

If a test throws a RuntimeException will not the test break?

---

Is there a reason for avoiding RunTimeExceptions in test cases?

---

I don't think to throw a JMSException here is suitable since JMSExeptions are intended to be used with JMS provider related issue. The most appropriate one is IllegalStateException[2] since this is due to the method has been invoked at an illegal or inappropriate time. But then again IllegalStateException is a runtime exception.

Classes like ArrayList is also throwing exceptions IllegalStateException, IllegalArgumentException, and NullPointerException when it is called with incorrect arguments and invoked in inappropriate time. Therefore I think the check that we have here is quite unnecessary. 

[1] https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html
[2] https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalStateException.html